src/modules/navigation/hooks/useFileLink.tsx

```diff
@@ -99,11 +101,24 @@ const useFileLink = (
     ...Object.fromEntries(new URLSearchParams(to.search).entries())
   })
 
+  const generateSharedDrivePath = (
+    driveId: string,
+    folderId: string
+  ): string => {
+    return `#/shareddrive/${driveId}/${folderId}`
+  }
+
   // nextcloud-file is a special case because Nextcloud are not in cozy ecosystem
   // so we open their link directly
   const href =
     type === 'nextcloud-file'
       ? path
+      : file.dir_id === SHARED_DRIVES_DIR_ID
+      ? generateSharedDrivePath(
+          // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+          file.relationships?.referenced_by?.data?.[0]?.id as string,
+          file._id
+        )
```

Comment:
Did you try to implement this (which is correct) in the `computeFileType` and `computePath` method. It's not a special case as nextcloud.